src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -82,12 +83,26 @@ const SidebarSidecarLayoutContent = ({
     },
   }
 
+  /**
+   * Handles the top positioning of the sidebar if a user has scrolled
+   * and prevents scrolling on the rest of the page body
+   */
+  useEffect(() => {
+    if (sidebarIsOpen) {
+      setScrollDelta(window.scrollY)
+      document.body.style.overflow = 'hidden'
```

Comment:
Open to a suggestion for a better way to handle this 😸

Reply:
Was just coming here to share what Bryce did for the search component! I can't remember where we chatted, but he said this code would be good to lift up to be reusable: https://github.com/hashicorp/dev-portal/blob/main/src/components/algolia-search/components/panel.tsx#L97-L109

```jsx
/**
 * Ensures the body can't be scrolled while the search panel is open on mobile viewports.
 */
useEffect(() => {
  const initialValue = document.body.style.overflow
  if (!isDesktop) {
    document.body.style.overflow = 'hidden'
  }

  return () => {
    document.body.style.overflow = initialValue
  }
}, [isDesktop])
```

Reply:
Ah cool! I'll lift this up then, ty 😄

Reply:
Can we use position fixed on the sidebar and overflow hidden on the body?  That'd reduce the need to do any scroll page offset calculations.

https://user-images.githubusercontent.com/825855/169607308-97afb2ad-fef1-4c09-95a6-82e702578231.mp4

Reply:
`--sticky-bars-height` doesn't account for text wrapping, which is why the top of the sidebar is a little cut off.

Reply:
I think the alpha callout with the wrapping text is going away for beta, but I'm not 100% sure. @BrandonRomano do you know? We will probably be able to make `--sticky-bars-height` less brittle and have less maintenance to do on it if so!